pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/memory/unsafe/UnsafePinotBufferFactory.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.memory.unsafe;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteOrder;
+import org.apache.pinot.segment.spi.memory.NonNativePinotDataBuffer;
+import org.apache.pinot.segment.spi.memory.PinotBufferFactory;
+import org.apache.pinot.segment.spi.memory.PinotDataBuffer;
+import xerial.larray.impl.OSInfo;
+
+
+public class UnsafePinotBufferFactory implements PinotBufferFactory {
+  public UnsafePinotBufferFactory() {
+    if (OSInfo.isWindows()) {
```

Comment:
Does this mean WINDOWS with java 17 won't work?

Reply:
We are using OS dependent things apis here and we didn't tried them in Windows. Reading the code of net.openhft.chronicle.core, it seems Windows has its own particularities when dealing with memory mapped files. In case we don't fulfill with them, they can just kill the JVM.

If you prefer we can just print a warning, but I think it is safer to just do not support Windows with Java 17/21.